emcache/client.py

```diff
@@ -866,7 +924,9 @@ async def create_client(
         autodiscovery_timeout,
     )
 
+    # sasl authorization via nodes
+    await client.auth(username, password)
```

Comment:
Maybe Im getting it wrong, but unless Im missing something we need to move this authentication so somewhere were every time that a new connection is created it gets authenticated. This will work only for the first connection that is created but other connections that are made will need to also get authenticated.

IMO we might need to move this to for example to [here](https://github.com/emcache/emcache/blob/master/emcache/connection_pool.py#L231) so every time that a new connection is created we also authenticate the connection, if can not be authenticated either we can just signal this as an error and we do not persist the connection or we just move ahead and any command that will be issued using this connection would fail.

The other option is to wait for making the authentication for the first command sent by the user, the benefits of doing this is that we can bubble up to the caller an authentication error, but the changes could be a bit more complicated. Also subsequent client calls using the same connection would either try again to make the authentication - if the previous failed, or they would just bypass the auth round trip since this was already made.

TBH i would be more in favour of the second option if possible since it signals easily to the user that there was an authentication issue (we can have an specific exception for this one)

Thoughts? or am I missed something in your PR?

Reply:
This is indeed true, and I will merge the changes with unix PR.

Reply:
@pfreixes, do you think this should work at the create_protocol level or at the create_connection level?

Reply:
Thank you very much for the clarification.

Reply:
@pfreixes, What do you think about this? I don't want to raise any exceptions at the python level. That is, when calling any command, the server will return an authentication error and python will simply call Command Error with information about the authentication error, which is correct! In principle, you told me about it more than once, apparently I didn't understand right away. This is both conceptually correct and does not break the library code, everything will remain as it is only with an authentication attempt when creating a connection!